.github/workflows/server-ci-master.yml

```diff
@@ -5,9 +5,27 @@ on:
       - master
       - cloud
       - release-*
-      - mono-repo*
 
 jobs:
   master-ci:
     uses: ./.github/workflows/server-ci-template.yml
     secrets: inherit
+  on-failure:
+    runs-on: ubuntu-latest
+    needs: master-ci
+    if: ${{ always() && needs.master-ci.result == 'failure' }}
+    steps:
+      - name: Report failure
+        env:
+          REPOSITORY: ${{ github.repository }}
+          HEAD_BRANCH: ${{ github.event.workflow_run.head_branch }}
```

Comment:
This wouldn't work. The [`github.event.workflow_run.*` event data](https://docs.github.com/en/webhooks/webhook-events-and-payloads#workflow_run) is only available in a `workflow_run`-triggered workflow, but after this change the context for this workflow is within a `push` event.
Let me check the docs for the correct equivalent interpolation

Reply:
Ah true, as [Antonis mentions](https://github.com/mattermost/mattermost/pull/25366#pullrequestreview-1720337002) the remediation of simply putting the untrusted interpolations within intermediary variables should be all we need, according to the Jira ticket.
I think we can still keep the failure report within its `workflow_run` file, in this case